modules/k8s/errors.go

```diff
@@ -68,7 +68,7 @@ type PodNotAvailable struct {
 
 // Error is a simple function to return a formatted error message as a string
 func (err PodNotAvailable) Error() string {
-	return fmt.Sprintf("Pod %s is not available, reason: %s, message: %s", err.pod.Name, err.pod.Status.Reason, err.pod.Status.Message)
+	return fmt.Sprintf("Pod %s is not available, reason: %s, message: %s, phase: %s, conditions: %+v", err.pod.Name, err.pod.Status.Reason, err.pod.Status.Message, err.pod.Status.Phase, err.pod.Status.Conditions)
```

Comment:
I was wondering if can be added tests to track generated errors

Reply:
You mean Unit Tests or integration tests? Unit tests for sure I can add some.
For integration tests I'll need a bit of hand holding (at least pointers to similar tests) as I don't seem to get how terratest is e2e tested.

Reply:
Just added a bunch of tests. I've only reworked things a little to make less verbose and able to detect more complex failures.

Reply:
@denis256 any input here?